src/Checkbox.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import PropTypes from 'prop-types'
+import { Icon, Flex, Box, Label, theme } from '..'
+
+class Checkbox extends React.Component {
+  handleClick(e) {
+    const { name, value, onClick } = this.props
+    const checked = e.target.checked
+    onClick(name, value, checked)
+  }
+
+  render() {
+    const {
+      name,
+      id,
+      disabled,
+      size,
+      fontSize,
+      children,
+      onClick,
+      checked,
+      defaultChecked,
+      theme: themeProp,
+      ...restProps
+    } = this.props
+    const currentTheme = themeProp || theme
+    return (
+      <StyledLabel
+        htmlFor={name}
+        data-test="design-system-checkbox"
+        disabled={disabled}
+      >
+        <input
+          type="checkbox"
+          name={name}
+          id={this.props.id || name}
+          disabled={disabled}
+          defaultChecked={checked || defaultChecked}
+          {...restProps}
+          onChange={e => this.handleClick(e)}
+          style={{ opacity: 0, zIndex: -1, position: 'absolute' }}
+        />
+        <StyledCheckbox
+          data-test="checkbox-facade"
+          disabled={disabled}
+          size={size}
+        >
+          <Icon
+            name="check"
+            size={size - 4}
+            color={currentTheme.colors.white}
+          />
+        </StyledCheckbox>
+        {children}
+      </StyledLabel>
```

Comment:
Similar to the Radio component in #139 it'd be great to keep the label element separate so that the Checkbox component is more composable, e.g.:

```jsx
<Label>
  <Checkbox {...checkboxProps} />
  Check it
</Label>
```